### ISSUE TITLE
Invalid github.com/go-delve/liner version

### ISSUE THREAD

**Issue by pboguslawski** [go-delve/delve#3025]
https://github.com/go-delve/delve/blob/9c5777e76218212f3790354f9470fb907e66cacb/go.mod#L10

There is no v1.2.2-1 tag in

https://github.com/go-delve/liner/tags

**Comment by pboguslawski** [comment]
It's a branch so it's ok - sorry for dirsturbing.

**Comment by pboguslawski** [comment]
Seems that after https://github.com/golang/go/issues/51312#issuecomment-1047875925 go does not treat branches with semantic-version names as releases.

go.mod points now to github.com/go-delve/liner version v1.2.2-1 but is a branch name not tag; this causes go 1.18 errors like
```
$ export GOPROXY=direct
$ go get github.com/go-delve/liner@v1.2.2-1
go: github.com/go-delve/liner@v1.2.2-1: invalid version: resolves to version v1.2.3-0.20220127212407-d32d89dd2a5d (v1.2.2-1 is not a tag)
```
Using default goproxy does not return error (maybe tag was cached before changes):
```
$ export GOPROXY=https://proxy.golang.org
$ go get github.com/go-delve/liner@v1.2.2-1
go: downloading github.com/go-delve/liner v1.2.2-1
go: downloading github.com/mattn/go-runewidth v0.0.3
go: downloading golang.org/x/sys v0.0.0-20211117180635-dee7805ff2e1
go: added github.com/go-delve/liner v1.2.2-1
go: added github.com/mattn/go-runewidth v0.0.3
go: added golang.org/x/sys v0.0.0-20211117180635-dee7805ff2e1
```
Please verify and fix.

**Comment by bcmills** [comment]
> Seems that after https://github.com/golang/go/issues/51312#issuecomment-1047875925 go does not treat branches with semantic-version names as releases.

To be precise, that behavior was changed for golang/go#35671, which was also backported to Go 1.16 and Go 1.17 as a minor security issue.
